service/frontend/api/handler_test.go

```diff
@@ -2310,6 +2310,167 @@ func (s *workflowHandlerSuite) TestRespondActivityTaskCanceled() {
 
 }
 
+func (s *workflowHandlerSuite) TestRespondDecisionTaskCompleted() {
+	validRequest := &types.RespondDecisionTaskCompletedRequest{
+		TaskToken: []byte("token"),
+		Identity:  "identity",
+		Decisions: make([]*types.Decision, 100),
+	}
+	mockResp := &types.HistoryRespondDecisionTaskCompletedResponse{
+		StartedResponse: &types.RecordDecisionTaskStartedResponse{
+			Attempt:          1,
+			ScheduledEventID: 2,
+		},
+	}
+	config := s.newConfig(dc.NewInMemoryClient())
+	config.EnableClientVersionCheck = dc.GetBoolPropertyFn(true)
+	wh := NewWorkflowHandler(s.mockResource, config, s.mockVersionChecker, nil)
+	wh.tokenSerializer = s.mockTokenSerializer
+
+	testInput := map[string]struct {
+		input       *types.RespondDecisionTaskCompletedRequest
+		mockFn      func()
+		expectError bool
+	}{
+		"shutting down": {
+			input: validRequest,
+			mockFn: func() {
+				wh.shuttingDown = int32(1)
+			},
+			expectError: true,
+		},
+		"nil request": {
+			input:       nil,
+			mockFn:      func() {},
+			expectError: true,
+		},
+		"nil task token": {
+			input: &types.RespondDecisionTaskCompletedRequest{
+				TaskToken: nil,
+			},
+			mockFn:      func() {},
+			expectError: true,
+		},
+		"deserialization failure": {
+			input: validRequest,
+			mockFn: func() {
+				s.mockTokenSerializer.EXPECT().Deserialize(gomock.Any()).Return(nil, errors.New("failed to deserialize token"))
+			},
+			expectError: true,
+		},
+		"empty domain ID": {
+			input: validRequest,
+			mockFn: func() {
+				s.mockTokenSerializer.EXPECT().Deserialize(gomock.Any()).Return(&common.TaskToken{DomainID: ""}, nil)
+			},
+			expectError: true,
+		},
+		"cannot get domain name": {
+			input: validRequest,
+			mockFn: func() {
+				s.mockTokenSerializer.EXPECT().Deserialize(gomock.Any()).Return(&common.TaskToken{DomainID: s.testDomainID}, nil)
+				s.mockDomainCache.EXPECT().GetDomainName(s.testDomainID).Return("", errors.New("error getting domain name"))
+			},
+			expectError: true,
+		},
+		"exceeds id length limit": {
+			input: validRequest,
+			mockFn: func() {
+				s.mockTokenSerializer.EXPECT().Deserialize(gomock.Any()).Return(&common.TaskToken{DomainID: s.testDomainID}, nil)
+				s.mockDomainCache.EXPECT().GetDomainName(s.testDomainID).Return(s.testDomain, nil)
+				wh.config.MaxIDLengthWarnLimit = dc.GetIntPropertyFn(1)
+				wh.config.IdentityMaxLength = dc.GetIntPropertyFilteredByDomain(1)
+			},
+			expectError: true,
+		},
+		"exceeds decision size limit": {
+			input: validRequest,
+			mockFn: func() {
+				s.mockTokenSerializer.EXPECT().Deserialize(gomock.Any()).Return(&common.TaskToken{DomainID: s.testDomainID}, nil)
+				s.mockDomainCache.EXPECT().GetDomainName(s.testDomainID).Return(s.testDomain, nil)
+				wh.config.DecisionResultCountLimit = dc.GetIntPropertyFilteredByDomain(10)
+			},
+			expectError: true,
```

Comment:
Can we assert error message? You can change through `expectError string` and `assert.ErrorContains()`

Reply:
Thank you for your comment, I have made the change.